kratos/includes/serializer.h

```diff
@@ -1152,7 +1162,11 @@ class KRATOS_API(KRATOS_CORE) Serializer
 
         rData.resize(size1,size2);
 
+#ifdef KRATOS_USE_AMATRIX   // This macro definition is for the migration period and to be removed afterward please do not use it 
+        read(rData.data(), rData.data() + rData.size(),0);
```

Comment:
maybe a stupid question but ... can't we have AMatrix rData to have the end?

Reply:
Then you expect a iterator not a pointer..... But definitely possible.

I decided to give it the same interface as pointer which is.

Reply:
TDataType* is actually the iterator for a c array, isn't it?